build-system/app.js

```diff
@@ -1342,7 +1342,7 @@ function addViewerIntegrationScript(ampJsVersion, file) {
 }
 
 function getUrlPrefix(req) {
-  return req.protocol + '://' + req.headers.host;
+  return '//' + req.headers.host;
```

Comment:
@zhouyx Can you review this change since you originally added this code?

Reply:
: ) I don't think I still remember why we need this `getUrlPrefix()` method. The code was from three years ago. Could you please briefly explain why we don't need the `req.protocol` anymore? If any I think we could remove the method and use relative url instead?

Reply:
We don't need the protocol because if we omit it, then the browser will just re-use the protocol of the current page. So it is a protocol-relative URL.

Reply:
Thanks. In this case can we also get rid of the `host`?

Reply:
Probably! 😄 It didn't occur to me to go that far. I'll have to try it.

Reply:
OK, I checked and eliminating the `host` won't work without making additional changes. The local proxied pages do indeed include scripts referenced like this:

```html
<script async src=/dist/amp.js type=text/javascript></script>
```

But then if I try accessing a proxied URL like at `https://amphtml.lndo.site/proxy/s/amp-wp.org/` then they end up 404ing:

![image](https://user-images.githubusercontent.com/134745/63220292-ea047200-c138-11e9-88ec-dcf90b658fca.png)

This because the base URL is changed in the proxied output via:

```html
<base href="https://cdn.ampproject.org/">
```

This is being done in `proxyToAmpProxy()`:

https://github.com/ampproject/amphtml/blob/d803fb31ba92b8d1fb55f08457b19ae06ec01522/build-system/app.js#L471-L472

Reply:
I see. Thanks for the detailed explanation.  LGTM

Reply:
Cherry-picked to another PR: #24066.